rust/polars-ocaml/src/sql_context.rs

```diff
@@ -43,6 +43,34 @@ fn rust_sql_context_register(
     OCaml::unit()
 }
 
+#[ocaml_interop_export]
+fn rust_sql_context_execute_with_data_frames(
+    cr: &mut &mut OCamlRuntime,
+    data_frames: OCamlRef<OCamlList<DynBox<crate::data_frame::PolarsDataFrame>>>,
+    names: OCamlRef<OCamlList<String>>,
+    query: OCamlRef<String>,
+) -> OCaml<Result<DynBox<crate::data_frame::PolarsDataFrame>, String>> {
+    let data_frames: Vec<crate::data_frame::PolarsDataFrame> = 
+        unwrap_abstract_vec(data_frames.to_rust(cr));
+    let names: Vec<String> = names.to_rust(cr);
+    let query: String = query.to_rust(cr);
+
+    let mut sql_context = SQLContext::new();
+    let _ = data_frames
+        .into_iter()
+        .zip(names.iter())
+        .map(|(df, name)| sql_context.register(&name, df.borrow().clone().lazy()))
+        .collect::<Vec<_>>();
+
+    match sql_context.execute(&query) {
```

Comment:
Does `sql_context.execute(&query).map(...)` not work here?

Reply:
I think (?) I want a bind here? I tried using map to do this and I wasn't able to figure out how.

Reply:
Ah my bad, I agree you want a bind. IIRC Rust's bind equivalent is flat_map. Does that work?

Reply:
done (used and_then)